packages/java/endpoint/src/main/java/com/vaadin/hilla/EndpointController.java

```diff
@@ -196,13 +241,13 @@ public ResponseEntity<String> serveEndpoint(
                             EndpointAccessChecker.ACCESS_DENIED_MSG));
         }
 
+        DAUUtils.EnforcementResult enforcementResult = null;
         try {
-            // Put a VaadinRequest in the instances object so as the request is
-            // available in the endpoint method
-            VaadinServletService service = (VaadinServletService) VaadinService
-                    .getCurrent();
-            CurrentInstance.set(VaadinRequest.class,
-                    new VaadinServletRequest(request, service));
+            enforcementResult = DAUUtils.trackDAU(this.vaadinService, request,
+                    response);
+            if (enforcementResult.isEnforcementNeeded()) {
+                return buildEnforcementResponseEntity(enforcementResult);
+            }
```

Comment:
Does it make sense to run this block only if `DAUUtils.isDauEnabled(service)` is true?

Reply:
`VaadinRequest` wouldn't be set to `CurrentInstance` then and cleared afterwards. `DAUUtils.trackDAU` does that now. It was set there originally, so I would assume something to break in Hilla without it.